retrybp/doc_test.go

```diff
@@ -85,3 +86,70 @@ func Example() {
 	// TODO: use result
 	_ = result
 }
+
+// This example demonstrates how to use retrybp with breakerbp.
+func ExampleBreakerErrorFilter() {
+	const timeout = time.Millisecond * 200
+
+	// TODO: use the actual type of your operation.
+	type resultType = int
+
+	// In real code this should be your actual operation you need to retry.
+	// Here we just use a placeholder for demonstration purpose.
+	operation := func() (resultType, error) {
+		return 0, nil
+	}
+
+	breaker := breakerbp.NewFailureRatioBreaker(breakerbp.Config{
+		// TODO: Add breaker configs.
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	const defaultRetries = 2
+	var result resultType
+	err := retrybp.Do(
+		ctx,
+		func() error {
+			// This is the main function call,
+			// wrapped with breaker.Execute to apply circuit breaker on it.
+			res, err := breaker.Execute(func() (interface{}, error) {
+				ret, err := operation()
+				return ret, err
+			})
+			result, _ = res.(resultType)
+			return err
+		},
+
+		// Default retry options, can be overridden by ctx passed in.
+		retry.Attempts(defaultRetries+1), // retries + the first attempt
+		retrybp.CappedExponentialBackoff(retrybp.CappedExponentialBackoffArgs{
+			InitialDelay: time.Millisecond * 10,
+			MaxJitter:    time.Millisecond * 5,
+		}),
+		retrybp.Filters(
```

Comment:
🔕 "Filters" here doesn't explain to me what it does, honestly.  Is this "Retry if it _matches_" or "Retry _except_ if it matches"? I can't tell without looking at the docs.

Reply:
Neither. Filters make individual decisions (it could be both), and defer to the next filter in the chain if they cannot make the decision.

Those are explained by the doc of Filters and Filter. I don't think it makes sense to repeat the same doc here. When people are looking at examples they are already on pkg.go.dev so I think it's reasonable to expect them to look at the doc of the types/functions if they want to learn more.